sharding/node/backend.go

```diff
@@ -104,14 +107,34 @@ func New(ctx *cli.Context) (*ShardEthereum, error) {
 
 // Start the ShardEthereum service and kicks off the p2p and actor's main loop.
 func (s *ShardEthereum) Start() {
+
 	log.Info("Starting sharding node")
+
 	for kind, service := range s.services {
 		// Start the next service.
 		if err := service.Start(); err != nil {
 			log.Error(fmt.Sprintf("Could not start service: %v, %v", kind, err))
 			s.Close()
 		}
 	}
+
+	go func() {
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
+		defer signal.Stop(sigc)
+		<-sigc
+		log.Info("Got interrupt, shutting down...")
+		go s.Close()
+		for i := 10; i > 0; i-- {
+			<-sigc
+			if i > 1 {
+				log.Warn("Already shutting down, interrupt more to panic.", "times", i-1)
+			}
+		}
```

Comment:
You need to do something after exiting this for loop, like a `panic()` or similar. This happens when the user has sent 10+ SIGINT to the process.

See geth's example: https://github.com/ethereum/go-ethereum/blob/6cf0ab38bd0af77d81aad4c104979cebee9e3e63/cmd/utils/cmd.go#L83